pilot/pkg/trustbundle/trustbundle.go

```diff
@@ -161,19 +177,103 @@ func (tb *TrustBundle) UpdateTrustAnchor(anchorConfig *TrustAnchorUpdate) error
 	return nil
 }
 
+func (tb *TrustBundle) updateRemoteEndpoint(spiffeEndpoints []string) {
+	tb.mutex.RLock()
+	remoteEndpoints := tb.endpoints
+	tb.mutex.RUnlock()
+
+	endpointChange := !isEqSpliceStr(spiffeEndpoints, remoteEndpoints)
+	if !endpointChange {
+		return
+	}
+	trustBundleLog.Infof("updated trustAnchor Remote Endpoint  :%v", spiffeEndpoints)
+
+	tb.mutex.Lock()
+	tb.endpoints = spiffeEndpoints
+	tb.endpointUpdateChan <- struct{}{}
+	tb.mutex.Unlock()
+}
+
 // AddMeshConfigUpdate : Update trustAnchor configurations from meshConfig
 func (tb *TrustBundle) AddMeshConfigUpdate(cfg *meshconfig.MeshConfig) {
+	var err error
 	if cfg != nil {
 		certs := []string{}
+		endpoints := []string{}
 		for _, pemCert := range cfg.GetCaCertificates() {
-			certs = append(certs, pemCert.GetPem())
+			cert := pemCert.GetPem()
+			if cert != "" {
+				certs = append(certs, cert)
+			} else if pemCert.GetSpiffeBundleUrl() != "" {
+				endpoints = append(endpoints, pemCert.GetSpiffeBundleUrl())
+			}
 		}
-		err := tb.UpdateTrustAnchor(&TrustAnchorUpdate{
+
+		err = tb.UpdateTrustAnchor(&TrustAnchorUpdate{
```

Comment:
feels a bit weird we need to update trust anchor and update endpoints separately. Should we just make TrustAnchorUpdate accept certs and endpoints?